contentmgr/replication.go

```diff
@@ -841,24 +809,42 @@ func (cm *ContentManager) addContentToStagingZone(ctx context.Context, content u
 	cm.BucketLk.Lock()
 	defer cm.BucketLk.Unlock()
 
-	blist, ok := cm.Buckets[content.UserID]
-	if !ok {
-		b, err := cm.newContentStagingZone(content.UserID, content.Location)
+	var zones []util.Content
+	if err := cm.DB.Find(&zones, "not active and pinning and aggregate and user_id = ?", content.UserID).Error; err != nil {
+		return nil
+	}
+	if len(zones) == 0 {
+		zone, err := cm.newContentStagingZone(content.UserID, content.Location)
 		if err != nil {
 			return fmt.Errorf("failed to create new staging zone content: %w", err)
 		}
 
-		_, err = cm.tryAddContent(b, content)
+		_, err = cm.tryAddContent(*zone, content)
 		if err != nil {
 			return fmt.Errorf("failed to add content to staging zone: %w", err)
 		}
-
-		cm.Buckets[content.UserID] = []*ContentStagingZone{b}
 		return nil
 	}
 
-	for _, b := range blist {
-		ok, err := cm.tryAddContent(b, content)
+	// find zones without enough capacity and exclude them as zone options
+	var inviableZoneIDs []uint
```

Comment:
no need for this if we can do https://github.com/application-research/estuary/pull/609/files#r1036184580